blockchain/difficulty.go

```diff
@@ -712,13 +712,18 @@ func (b *BlockChain) calcNextRequiredStakeDifficulty(curNode *blockNode) (int64,
 }
 
 // CalcNextRequiredStakeDifficulty calculates the required stake difficulty for
-// the block after the end of the current best chain based on the active stake
-// difficulty retarget rules.
+// the block after the given block based on the active stake difficulty retarget
+// rules.
 //
 // This function is safe for concurrent access.
-func (b *BlockChain) CalcNextRequiredStakeDifficulty() (int64, error) {
+func (b *BlockChain) CalcNextRequiredStakeDifficulty(hash *chainhash.Hash) (int64, error) {
```

Comment:
This code change looks good to me, but is this exported `CalcNextRequiredStakeDifficulty` function even needed at this point since everything else was updated to use the best state to get the next required stake difficulty?

Reply:
Not right now, but I expect it will be needed with the out of order decoupled logic.

Reply:
Makes sense.